core/src/backend/navigator.rs

```diff
@@ -113,6 +114,23 @@ pub struct Response {
     pub body: Vec<u8>,
 }
 
+/// An error response from a fetch request
+#[derive(Debug, Error)]
+pub enum FetchError {
```

Comment:
I'd rather just use `Error::FetchError()` with an appropriate string. The error type doesn't really matter, AFAIK it's only used for logging.

Reply:
This actually is used now in loader.rs (see line 1310 for example), where it influences which listeners are invoked.